modules/login/src/main/frontend/src/login/loginForm.js

```diff
@@ -145,114 +128,128 @@ class SignIn extends React.Component {
+                <InputLabel htmlFor="j_username">Username{singleStepEntry ? "" : " or email address"}</InputLabel>
+                <Input
+                  id="j_username"
+                  name="j_username"
+                  autoComplete="email"
+                  autoFocus
+                  onChange={(event) => setUsername(event.target.value)}
+                />
+              </FormControl>
+              {  (!singleStepEntry) &&
+                <Button
+                  fullWidth
+                  variant="contained"
+                  className={`${classes.actions} ${classes.submit}`}
+                  onClick={nextButtonCallback}
+                  disabled={username.length == 0}
+                >
+                  Next
+                </Button>
+              }
+            </React.Fragment>
+          }
 
-            <form
-              method="post"
-              className={classes.form}
-              onSubmit={(event)=> {
-                event.preventDefault();
-                if (this.state.phase == "PASSWORD_ENTRY" || this.state.singleStepEntry === true) {
-                  this.submitLogin();
-                } else if (this.state.phase == "USERNAME_ENTRY" && this.state.singleStepEntry === false) {
-                  nextButtonCallback();
-                }
-              }}
-            >
-              { (this.state.phase == "USERNAME_ENTRY" || this.state.singleStepEntry) &&
-                <React.Fragment>
-                  <FormControl variant="standard" margin="normal" required fullWidth>
-                    <InputLabel htmlFor="j_username">Username{this.state.singleStepEntry ? "" : " or email address"}</InputLabel>
-                    <Input id="j_username" name="j_username" autoComplete="email" autoFocus onChange={(event) => {this.setState({username: event.target.value});}}/>
-                  </FormControl>
-                  {  (!this.state.singleStepEntry) &&
+          { (phase == "PASSWORD_ENTRY" || singleStepEntry) &&
+            <React.Fragment>
+              <FormControl variant="standard" margin="normal" required fullWidth>
+                <InputLabel htmlFor="j_password">Password{singleStepEntry ? "" : (" for " + username)}</InputLabel>
+                <Input
+                  name="j_password"
+                  type={passwordIsMasked ? 'text' : 'password'}
+                  id="j_password" autoComplete="current-password"
+                  autoFocus={phase === "PASSWORD_ENTRY"}
+                  onChange={(event) => setPassword(event.target.value)}
+                  endAdornment={
+                    <InputAdornment position="end">
+                      <Tooltip title={passwordIsMasked ? "Mask Password" : "Show Password"}>
+                        <IconButton
+                          size="large"
+                          aria-label="Toggle password visibility"
+                          onClick={togglePasswordMask}
```

Comment:
I would get rid of the `togglePasswordMask` function since it's only used here.

```suggestion
                          onClick={() => {setPasswordIsMasked(!passwordIsMasked)}}
```